x-pack/elastic-agent/pkg/artifact/download/fs/downloader.go

```diff
@@ -42,7 +42,7 @@ func NewDownloader(config *artifact.Config) *Downloader {
 func (e *Downloader) Download(_ context.Context, programName, version string) (string, error) {
 	// create a destination directory root/program
 	destinationDir := filepath.Join(e.config.TargetDirectory, programName)
-	if err := os.MkdirAll(destinationDir, os.ModeDir); err != nil {
+	if err := os.MkdirAll(destinationDir, 0755); err != nil {
```

Comment:
This is a drive-by fix on Mac `os.ModeDir` does not create the directory with the proper permissions. `0755` must be used.